extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/BridgeHandler.java

```diff
@@ -199,8 +200,12 @@ private Config loadAndCheckConfig() {
         if (StringUtils
                 .isNotBlank(thingConfig.get(DigitalSTROMBindingConstants.SENSOR_DATA_UPDATE_INTERVAL).toString())) {
             try {
-                config.setSensordataRefreshInterval(Integer.parseInt(
-                        thingConfig.get(DigitalSTROMBindingConstants.SENSOR_DATA_UPDATE_INTERVAL).toString() + "000"));
+                Object param;
+                param = thingConfig.get(DigitalSTROMBindingConstants.SENSOR_DATA_UPDATE_INTERVAL);
+                logger.debug("param = {}", param);
+                if (param instanceof BigDecimal && param != null) {
+                    config.setSensordataRefreshInterval(((BigDecimal) param).intValue());
```

Comment:
The convertion from seconds to milliseconds ( * 1000) is missing. Also at the next two config parameters.